webapp/lib/controller.dart

```diff
@@ -169,59 +131,146 @@ void loadDashboardView() {
 
 void loadBatchRepliesConfigurationView() {
   view.contentView.renderView(new view.BatchRepliesConfigurationView(model.changeCommsPackage));
-  command(UIAction.loadBatchRepliesPackageConfiguration, null);
 }
 
 void loadEscalatesConfigurationView() {
   view.contentView.renderView(new view.EscalatesConfigurationView());
-  command(UIAction.loadEscalatesPackageConfiguration, null);
 }
 
 loadProjectConfigurationView() {
   view.contentView.renderView(new view.ProjectConfigurationView());
 }
 
-void fetchConfigurationData() {
-  configurationTagData = model.configurationData;
-  additionalConfigurationTags = model.configurationTags;
-  configurationResponseLanguages = model.configurationReponseLanguageData;
+// Tag Operations
+enum TagOperation {
+  ADD,
+  UPDATE,
+  REMOVE
+}
+
+void _addTag(String tag, model.TagStyle tagStyle, Map<String, model.TagStyle> tagType, [bool isEditable = false]) {
+  tagType.addAll({tag: tagStyle});
+  if (!isEditable) model.changeCommsPackage.availableTags.remove(tag);
+}
+
+Map<String, model.TagStyle> _updateTag(String originalTag, String updatedTag, Map<String, model.TagStyle> tagType) {
```

Comment:
can we make this function like the add and remove ones - ie. to update the `tagType` (`tagCollection`) in place rather than returning a new copy of it with the updated tag?

Reply:
Changed it to update `tagCollection` in-place